integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -1120,6 +1120,72 @@ def test_hash_groupby_collect_partial_replace_fallback(data_gen,
         non_exist_classes=','.join(non_exist_clz),
         conf=conf)
 
+# The special case is to test when the physical plan is being re-written due to the re-optimize
+# of AQE taking effect, which is rare in real world scenarios. So far, this kind of problem only
+# has encountered when there exists a local aggregate ahead of the TypedImperativeAggregate. Then,
+# the local aggregate will be folded by the original rule `FoldLocalAggregate`, which triggers the
+# re-write of the physical plan.
+#
+# In order to simulate this kind of scenario, we create a bucketed table and read it back to
+# enforce a local shuffle exchange and local aggregate before the TypedImperativeAggregate.
+# And wrap the columns of TypedImperativeAggregate with the expressions leading to fallbackToCpu(
+# `forall`), to create the AggregateExec half on CPU and half on GPU. Under this situation, when AQE
+# is enabled, the gpuToCpuBufferConverter being inserted into physical plan will be lost due to
+# the plan re-write, which requires us to store the gpuToCpuBufferConverter into related LogicalPlan
+# instead of PhysicalPlan only.
+@ignore_order(local=True)
+@allow_non_gpu('ObjectHashAggregateExec', 'SortAggregateExec',
+               'ShuffleExchangeExec', 'HashPartitioning', 'SortExec',
+               'SortArray', 'Alias', 'Literal', 'Count', 'CollectList', 'CollectSet',
+               'AggregateExpression', 'ProjectExec', *non_utc_allow)
+@pytest.mark.parametrize('data_gen', [[
+    ('k1', RepeatSeqGen(LongGen(), length=100)),
+    ('k2', RepeatSeqGen(LongGen(), length=20)),
+    ('v', StringGen(nullable=False))]], ids=idfn)
+@pytest.mark.parametrize('use_obj_hash_agg', ['false', 'true'], ids=idfn)
+@pytest.mark.parametrize("kudo_enabled", ["true", "false"], ids=idfn)
```

Comment:
Personally, only the default setting is good enough here for kudo.

Reply:
Removed the optional argument.